kubectl: warn when deletion is blocked by finalizers

elizhabiboullah opened the pull request:
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR improves the user experience of `kubectl delete` when a resource cannot be deleted immediately due to pending finalizers (e.g., AWS ALB controller resources, ExternalSecrets).
Currently, `kubectl delete` reports `"<resource>" deleted` (to stdout) even if the deletion is blocked by a finalizer. This leads to user confusion ("Why is it still there?") and breaks GitOps sync loops that believe the resource is gone.

#### Which issue(s) this PR is related to:
N/A

#### Special notes for your reviewer:

I used the `WarningPrinter` to ensure the message goes to `stderr` and follows standard warning formatting. A unit test has been added to verify the warning logic.

#### Does this PR introduce a user-facing change?

kubectl delete now prints a warning to stderr when a resource is deleted but remains pending due to finalizers. The warning includes a suggested `kubectl patch` command to remove the finalizers if needed

```
$ kubectl delete externalsecret my-secret
```
``` 
The resource "my-secret" has finalizers (external-secrets.io/finalizer) and will not be deleted until they are removed.
To remove them manually, run:
kubectl patch externalsecret my-secret --type=merge -p '{"metadata":{"finalizers":null}}'
externalsecret.external-secrets.io "my-secret" deleted
```

elizhabiboullah committed:
feat(kubectl): explain why deletion is blocked when finalizers are present

lmktfy commented on staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go:
```diff
@@ -509,6 +509,20 @@ func (o *DeleteOptions) PrintObj(info *resource.Info) {
+		if accessor, err := meta.Accessor(response); err == nil {
+			if accessor.GetDeletionTimestamp() != nil && len(accessor.GetFinalizers()) > 0 {
+				finalizers := strings.Join(accessor.GetFinalizers(), ", ")
+				msg := fmt.Sprintf(" The resource %q has finalizers (%s) and will not be deleted until they are removed.\n", info.Name, finalizers)
```
If there are 8 finalizers present, what should we print?

lmktfy reviewed (commented):
Thank you for the PR.

elizhabiboullah commented on staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go:
```diff
@@ -509,6 +509,20 @@ func (o *DeleteOptions) PrintObj(info *resource.Info) {
+			if accessor.GetDeletionTimestamp() != nil && len(accessor.GetFinalizers()) > 0 {
+				finalizers := strings.Join(accessor.GetFinalizers(), ", ")
+				msg := fmt.Sprintf(" The resource %q has finalizers (%s) and will not be deleted until they are removed.\n", info.Name, finalizers)
+				msg += fmt.Sprintf("    To remove them manually, run:\n    kubectl patch %s %s --type=merge -p '{\"metadata\":{\"finalizers\":null}}'", info.Mapping.Resource.Resource, info.Name)
```
Fair point, I'll drop the patch command

elizhabiboullah commented:
Hi @seans3 @soltysh @lmktfy 

I believe I've addressed the review comments, could someone help trigger the CI tests with /ok-to-test when convenient?

elizhabiboullah committed:
Merge branch 'kubernetes:master' into fix/kubectl-finalizer-feedback
